File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-1.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int length = nums.size()l
        int l = 1l
        for(int r = 1; r < length; r++) {
            if(num[r] != nums[r-1]) {
                nums[l++] = num[r];
            }
        }

        return l
    }
};

File: Data Structures & Algorithms/sum-of-all-subset-xor-totals/submission-1.py
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        # Using backtracking method to create all possible subset

        # result sum
        res = 0
        def backtrack(i, subset):
            nonlocal res

            # tracking sum in each subset
            xorr = 0
            for num in subset:
                xorr ^= num
            
            # sum back to result
            res += xorr

            # create a subset
            for j in range(i, len(nums)):
                subset.append(nums[j])
                backtrack(j+1, subset)
                subset.pop()
        
        # start from topdown
        backtrack(0, [])
        return res